lib/src/main/scala/spinal/lib/bus/simple/PipelinedMemoryBus.scala

```diff
@@ -42,6 +42,40 @@ case class PipelinedMemoryBus(config : PipelinedMemoryBusConfig) extends Bundle
   val cmd = Stream(PipelinedMemoryBusCmd(config))
   val rsp = Flow(PipelinedMemoryBusRsp(config))
 
+  def readRequestFire = cmd.fire && !cmd.write
+  lazy val formalContract = new Composite(this, "formalContract") {
```

Comment:
Is this contract means to the bus's contract? Why it should invariantCmd be created?

Reply:
Part of this is nailing down what exactly the semantics are for PMB cmd/rsp. I'm guessing a lot of perfectly functional code would trip up if the stream contract was applied with payload invariance on cmd directly -- when `write` is false it seems fine to just not drive `data` or  `mask`.